src/os/bluestore/Compression.cc

```diff
@@ -56,19 +56,34 @@ void Estimator::cleanup()
   uncompressed_size = 0;
   compressed_occupied = 0;
   compressed_size = 0;
+  compressed_area = 0;
   total_uncompressed_size = 0;
   total_compressed_occupied = 0;
   total_compressed_size = 0;
   actual_compressed = 0;
   actual_compressed_plus_pad = 0;
   extra_recompress.clear();
+  single_compressed_blob = true;
+  last_blob = nullptr;
 }
+
+void Estimator::configure(const WriteContext& wctx)
+{
+  target_blob_size = wctx.target_blob_size;
+}
+
 inline void Estimator::batch(const BlueStore::Extent* e, uint32_t gain)
 {
   const Blob *h_Blob = &(*e->blob);
   const bluestore_blob_t &h_bblob = h_Blob->get_blob();
   if (h_bblob.is_compressed()) {
+    if (last_blob) {
+      single_compressed_blob &= (h_Blob == last_blob);
```

Comment:
This is meant to be a bit operation? if so - I'd rather we have that bit named, and perform the logic of setting it explicitly.

Reply:
IIUC that's not a bitwise op. Hence IMO using bitwise &= operator is rather confusing

Reply:
No, it was intended to be a logical OR. Fixing.